### ISSUE TITLE
nginx.conf.template not regenerating nginx.conf

### ISSUE THREAD

**Issue by pritchardjonathan** [progrium/dokku#1668]
Hi,

I'm trying to override the default nginx configuration so I've placed a nginx.conf.template file in the root of my application (repro: https://github.com/pritchardjonathan/dokkuconfigproof). 

When I deploy I see:
```
-----> Cleaning up...
-----> Building dokkuconfigproof from herokuish...
-----> Adding BUILD_ENV to build environment...
-----> Node.js app detected
       
-----> Creating runtime environment
       
       NPM_CONFIG_LOGLEVEL=error
       NPM_CONFIG_PRODUCTION=true
       NODE_ENV=production
       NODE_MODULES_CACHE=true
       
-----> Installing binaries
       engines.node (package.json):  unspecified
       engines.npm (package.json):   unspecified (use default)
       
       Resolving node version (latest stable) via semver.io...
       Downloading and installing node 5.0.0...
       Using default npm version: 3.3.6
       
-----> Restoring cache
       Skipping cache restore (new runtime signature)
       
-----> Building dependencies
       Pruning any extraneous modules
       Installing node modules (package.json)
       
-----> Caching build
       Clearing previous node cache
       Saving 2 cacheDirectories (default):
       - node_modules
       - bower_components (nothing to cache)
       
-----> Build succeeded!
       `-- (empty)
       
-----> Discovering process types
       Procfile declares types -> web
-----> Releasing dokkuconfigproof (dokku/dokkuconfigproof:latest)...
-----> Deploying dokkuconfigproof (dokku/dokkuconfigproof:latest)...
-----> DOKKU_SCALE file not found in app image. Generating one based on Procfile...
-----> New DOKKU_SCALE file generated
=====> web=1
-----> Running pre-flight checks
       For more efficient zero downtime deployments, create a file CHECKS.
       See http://progrium.viewdocs.io/dokku/checks-examples.md for examples
       CHECKS file not found in container: Running simple container check...
-----> Waiting for 120 seconds ...
-----> Default container check successful!
=====> dokkuconfigproof container output:
       Server listening on: http://localhost:8080
=====> end dokkuconfigproof container output
-----> Running post-deploy
=====> unsupported vhost config found. disabling vhost support
-----> Setting config vars
       NO_VHOST: 1
-----> Overriding default nginx.conf with detected nginx.conf.template
-----> Setting config vars
       NO_VHOST: 1
-----> no nginx port set. setting to random open high port
-----> Setting config vars
       DOKKU_NGINX_PORT: 18759
-----> Creating http nginx.conf
-----> Running nginx-pre-reload
       Reloading nginx
=====> Application deployed:
       http://xx.xx.xx.xx:32769 (container)
       http://xx.xx.xx.xx:18759 (nginx)
```

So I think it's recognising the template config but if I look at /home/dokkuconfigproof/nginx.conf after deploying the app, it doesn't match the template; just shows the default configuration:
```
upstream dokkuconfigproof {
  server 127.0.0.1:32769;
}
```

This is with dokku version 0.4.3 and file permissions looks fine:

```
drwxrwxr-x  2 dokku dokku 4096 Nov  7 02:23 branches
drwxrwxr-x  3 22666 22666 4096 Nov  7 02:27 cache
-rw-rw-r--  1 dokku dokku   66 Nov  7 02:23 config
-rw-rw-r--  1 dokku dokku   65 Nov  7 02:29 CONTAINER.web.1
-rw-rw-r--  1 dokku dokku   73 Nov  7 02:23 description
-rw-rw-r--  1 dokku dokku    6 Nov  7 02:27 DOKKU_SCALE
-rw-------  1 dokku dokku   52 Nov  7 02:29 ENV
-rw-rw-r--  1 dokku dokku   23 Nov  7 02:23 HEAD
drwxrwxr-x  2 dokku dokku 4096 Nov  7 02:23 hooks
drwxrwxr-x  2 dokku dokku 4096 Nov  7 02:23 info
-rw-rw-r--  1 dokku dokku   10 Nov  7 02:29 IP.web.1
-rw-------  1 dokku dokku   57 Nov  7 02:29 nginx.conf
-rw-r--r--  1 dokku dokku  120 Nov  7 02:23 nginx.conf.template
drwxrwxr-x 16 dokku dokku 4096 Nov  7 02:23 objects
-rw-rw-r--  1 dokku dokku    6 Nov  7 02:29 PORT.web.1
drwxrwxr-x  4 dokku dokku 4096 Nov  7 02:23 refs
-rw-rw-r--  1 dokku dokku   26 Nov  7 02:29 URL
```

I expect I'm missing something really obvious... any help much appreciated

Jon

**Comment by josegonzalez** [comment]
I *think* this is fixed by https://github.com/progrium/dokku/pull/1643, though we are currently blocked from releasing a package because of issues with our upstream provider packagecloud.io. Can you try your app on master?

**Comment by josegonzalez** [comment]
Okay I just released 0.4.4, can you try with that release?

**Comment by pritchardjonathan** [comment]
Yes upgrading to 0.4.4 solves the issue. Thanks for your help!
